torch/csrc/jit/mobile/tensor_hash.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <ATen/core/ivalue.h>
+
+namespace torch {
+namespace jit {
+
+struct tensor_value_hash {
+  std::size_t operator()(const at::Tensor& tensor) const {
+    std::stringstream tensor_stream;
+    tensor_stream << tensor;
```

Comment:
Curious where is operator<< defined? How long would the steam be?

Reply:
Good point. Previously I thought constant value might include other types, in addition to Tensor. For IValue, it's defined here. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/ivalue.cpp#L665. Probably it's better to find out the implementation for Tensor type. One example is:
```
torch._utils._rebuild_tensor_v2(pers.obj(('storage', torch.FloatStorage, '0', 'cpu', 90944),),
       0,
       (1, 116, 28, 28),
       (90944, 784, 28, 1),
       False,
       collections.OrderedDict())
```

Reply:
I think a tensor would be sufficient. Other ivalues does not take significant space. Is the example a tensor or tensor type?

Reply:
The example is a IValue and it's a tensor type. Let me check how to get it from Tensor.

Reply:
> I think a tensor would be sufficient. Other ivalues does not take significant space. Is the example a tensor or tensor type?

As I commented before, I dont see a reason to deduplicate only tensor constants. Seems like doing it selectively is adding the kind of overhead we can work around?

Reply:
Actually, when I check the constant in bytecode.pkl, there are some other types including int, bool, which are not available in constant from jit.

Reply:
> Curious where is operator<< defined? How long would the steam be?

It's defined here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/Formatting.cpp#L230. I was thinking to get the string directly, but it seems easier to just use the `std::stringstream`, such that export and import can be consistent.